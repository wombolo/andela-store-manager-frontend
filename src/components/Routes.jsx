import 'babel-polyfill';
import React from "react";
import { Route, Switch } from 'react-router-dom';

import Home from "./Home.jsx";
import Login from "./Login.jsx";
import PrivateRoute from "./PrivateRoute.jsx";
import NotFound from "./NotFound.jsx"
import Dashboard from "./Dashboard.jsx";
import AllProducts from "./AllProducts.jsx";
import AddProduct from "./AddProduct.jsx";
import ViewModifyProduct from "./ViewModifyProduct.jsx";
import Cart from "./Cart.jsx";
import AllSales from "./AllSales.jsx";
import ViewModifyProfile from "./ViewModifyProfile.jsx";
import AddProfile from "./AddProfile.jsx";

const Routes = () => (
  <Switch>
    <Route exact path="/" component={Home}> </Route>
    <Route exact path="/login" component={Login}/>
    <PrivateRoute path="/dashboard" component={Dashboard} />
    <PrivateRoute path="/all-products" component={AllProducts} />
    <PrivateRoute path="/add-product" component={AddProduct} />
    <PrivateRoute path="/add-profile" component={AddProfile} />
    <PrivateRoute exact path="/edit-product/:id" component={ViewModifyProduct} />
    <PrivateRoute exact path="/cart" component={Cart} />
    <PrivateRoute exact path="/all-sales" component={AllSales} />
    <PrivateRoute exact path="/profile/:id" component={ViewModifyProfile} />

    <Route component={NotFound} />
  </Switch>
);

export default Routes;
