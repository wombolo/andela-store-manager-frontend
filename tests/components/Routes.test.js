import 'babel-polyfill'
import React from 'react';
import { shallow } from 'enzyme';

import Routes from '../../src/components/Routes.jsx';

let wrapper;


describe('Routes Component', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<Routes />);
    expect(wrapper).toMatchSnapshot();
  });
});
