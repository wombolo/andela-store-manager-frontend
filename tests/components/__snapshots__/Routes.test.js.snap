// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component should match snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  >
     
  </Route>
  <Route
    component={[Function]}
    exact={true}
    path="/login"
  />
  <Connect(PrivateRoute)
    component={[Function]}
    path="/dashboard"
  />
  <Connect(PrivateRoute)
    component={[Function]}
    path="/all-products"
  />
  <Connect(PrivateRoute)
    component={[Function]}
    path="/add-product"
  />
  <Connect(PrivateRoute)
    component={[Function]}
    path="/add-profile"
  />
  <Connect(PrivateRoute)
    component={[Function]}
    exact={true}
    path="/edit-product/:id"
  />
  <Connect(PrivateRoute)
    component={[Function]}
    exact={true}
    path="/cart"
  />
  <Connect(PrivateRoute)
    component={[Function]}
    exact={true}
    path="/all-sales"
  />
  <Connect(PrivateRoute)
    component={[Function]}
    exact={true}
    path="/profile/:id"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
